packages/testnet-faucets/src/ts/handler.ts

```
@@ -62,37 +77,83 @@ export class Handler {
-        queueType: QueueType,
+        requestQueueByNetworkId: ItemByNetworkId<RequestQueue>,
+        requestedAssetType: RequestedAssetType,
     ) {
-        const recipientAddress = req.params.recipient;
-        if (_.isUndefined(recipientAddress) || !this._isValidEthereumAddress(recipientAddress)) {
-            res.status(400).send('INVALID_RECIPIENT_ADDRESS');
-            return;
-        }
-        const networkId = _.get(req.query, 'networkId', DEFAULT_NETWORK_ID);
-        const requestQueue = _.get(requestQueueByNetworkId, networkId);
+        const requestQueue = _.get(requestQueueByNetworkId, req.networkId);
         if (_.isUndefined(requestQueue)) {
-            res.status(400).send('INVALID_NETWORK_ID');
+            res.status(400).send('UNSUPPORTED_NETWORK_ID');
             return;
         }
-        const lowerCaseRecipientAddress = recipientAddress.toLowerCase();
-        const didAddToQueue = requestQueue.add(lowerCaseRecipientAddress);
+        const didAddToQueue = requestQueue.add(req.recipientAddress);
         if (!didAddToQueue) {
             res.status(503).send('QUEUE_IS_FULL');
             return;
         }
-        utils.consoleLog(`Added ${lowerCaseRecipientAddress} to queue: ${queueType} networkId: ${networkId}`);
+        utils.consoleLog(`Added ${req.recipientAddress} to queue: ${requestedAssetType} networkId: ${req.networkId}`);
         res.status(200).end();
     }
+    private async _dispenseOrder(req: express.Request, res: express.Response, requestedAssetType: RequestedAssetType) {
+        const zeroEx = _.get(this._zeroExByNetworkId, req.networkId);
+        if (_.isUndefined(zeroEx)) {
+            res.status(400).send('UNSUPPORTED_NETWORK_ID');
+            return;
+        }
+        res.setHeader('Content-Type', 'application/json');
+        const makerTokenAddress = await zeroEx.tokenRegistry.getTokenAddressBySymbolIfExistsAsync(requestedAssetType);
+        if (_.isUndefined(makerTokenAddress)) {
```

I feel like this conditional will only ever be hit if the developer messed up. For this reason, I think this should probably throw an error rather then returning one to the caller. Thoughts?

---

sure, makes sense